geniza/corpus/models.py

```diff
@@ -1287,10 +1290,15 @@ def merge_with(self, merge_docs, rationale, user=None):
         needs_review = [self.needs_review] if self.needs_review else []
 
         for doc in merge_docs:
-            # add merge id to old pgpid list
-            self.old_pgpids.append(doc.id)
             # add any tags from merge document tags to primary doc
+            # NOTE: This must be done before any other changes to self because it will fire
+            # m2m_changed signal which calls self.refresh_from_db()
```

Comment:
is this refresh from db likely to be problematic elsewhere? it seems like it could lead to unexpected behavior that might be hard to troubleshoot

in the signal handler, instead of refreshing, could we get a new instance from the db without modifying the local copy?

Reply:
Great idea, much better than refresh from db, thanks!